[STRMCMP-996] Adding translations for S3 and kinesis input sources

premsantosh opened the pull request:
Adding new translations for the S3 and kinesis source.
Also adding the composite transform that can be used to construct the S3 and Kinesis source.
------------------------
Once I build a version and tag it ill add a link here to an example running job.
**EDIT:** Example running job http://beamperfk8s5.ingress.semistateful-stg-0.us-east-1.k8s.lyft.net/#/job/55e396d43a225d0679e59c44b03cf018/overview

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [![Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [![Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [![Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [![Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [![Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[![Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[![Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [![Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [![Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [![Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [![Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [![Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [![Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [![Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/badge/icon)](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

maghamravi commented on runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java:
```diff
@@ -357,6 +375,216 @@ private void translateKinesisInput(
+      return occurredAtMs;
+    }
+
+    private static long parseDateTime(String datetime) {
```
recommend: I feel we should have a utility class (EventUtils?) with such helper methods to promote code reuse . I see a similar code at  (https://github.com/lyft/beam/blob/bd762d7f35b5b4f458e129b9ba719dcd5ba00299/runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java#L652)

premsantosh replied:
I agree, I was looking to use the parsing logic in EventUtils initially but it looks like its not currently serializable. I believe that would require some work so I've created https://jira.lyft.net/browse/STRMCMP-1109.

maghamravi commented on runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java:
```diff
@@ -357,6 +375,216 @@ private void translateKinesisInput(
+      ObjectWriter writer = mapper.writer();
+      byte[] bytes = new byte[0];
+      try {
+        bytes = writer.writeValueAsBytes(eventJson);
```
nit: can this be simplified to `mapper.writeValueAsBytes(event)`

maghamravi commented:
@premsantosh Can you add few tests for the new functionality.

premsantosh commented:
> @premsantosh Can you add few tests for the new functionality.

Yep, I've added tests for the java translations. Ill look to see if I can add anything for the PTransform on the python side but I don't believe there's much to test there. I was first trying to get a working version again before adding tests. I've also added a link to a test beam job.

maghamravi commented:
LGTM ! Nice work!!

maghamravi commented:
/approve (👍)

tweise commented on runners/flink/flink_runner.gradle:
```diff
@@ -152,6 +152,7 @@ dependencies {
   compile "org.apache.flink:flink-metrics-core:$flink_version"
   compile "org.apache.flink:flink-java:$flink_version"
   compile "org.apache.flink:flink-runtime_2.11:$flink_version"
+  compile("com.lyft:streamingplatform-events-source:2.1-dryft-SNAPSHOT")
```
This should be further down in the lyft specific block.

tweise replied:
Also, the size of the job server has grown significantly with this addition. Have you checked that we are only pulling in the minimum transitive dependencies?

tweise commented:
Commits should be squashed and tagged according to the existing convention. This is to reduce friction for upgrades.
